internal/server/iigointernal/executive.go

```diff
@@ -83,11 +83,13 @@ func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]share
 		if !e.incurServiceCharge(actionCost.BroadcastTaxationActionCost) {
 			return errors.Errorf("Insufficient Budget in common Pool: broadcastTaxation")
 		}
-		for _, islandID := range aliveIslands {
-			d := shared.CommunicationContent{T: shared.CommunicationInt, IntegerData: int(taxMapReturn.ResourceMap[islandID])}
-			data := make(map[shared.CommunicationFieldName]shared.CommunicationContent)
-			data[shared.TaxAmount] = d
-			communicateWithIslands(islandID, shared.TeamIDs[e.PresidentID], data)
+		for islandID, resourceAmount := range taxMapReturn.ResourceMap {
+			if Contains(aliveIslands, islandID) {
+				d := shared.CommunicationContent{T: shared.CommunicationInt, IntegerData: int(resourceAmount)}
+				data := make(map[shared.CommunicationFieldName]shared.CommunicationContent)
+				data[shared.TaxAmount] = d
+				communicateWithIslands(islandID, shared.TeamIDs[e.PresidentID], data)
```

Comment:
This means that any island not in the resource map is not told of how much tax they need to pay. This is more of a design issue so if that's the intended behaviour then it's fine.

Reply:
I'm unsure of how exactly design intended this, but I can imagine a lot of agent implementations may rely on some form of tax communication even if it is 0.